enumerable.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    to_a.length.times { |i| yield self[i] }
+    self
+  end
+
+  def my_each_with_index
+    to_a.my_each { |each| yield each, index(each) }
+    self
+  end
+
+  def my_select
+    array = []
+    to_a.my_each { |each| array.push(each) if yield each }
+    array
+  end
+
+  def my_all?(parameter = nil)
+    to_a.my_each do |each|
+      if block_given?
+        return false unless yield each
+      elsif parameter.instance_of? Class
+        return false unless each.is_a? parameter
+      else
+        return false unless each
+        return false unless parameter === each
+      end
+    end
+    true
+  end
+
+  def my_any?(parameter = nil)
+    to_a.my_each do |each|
+      if block_given?
+        return true if yield each
+      elsif parameter.instance_of? Class
+        return true if each.is_a? parameter
+      else
+        return true if each == true
+        return true if parameter === each
+      end
+    end
+    false
+  end
```

Comment:
## my_any
- [ ] when no block or argument is given returns false if at least one of the collection is not true
```ruby
false_array.my_any?
```
expected false
             got true